modules/react/modal/lib/hooks/useCloseOnOverlayClick.ts

```diff
@@ -30,14 +31,15 @@ export const useCloseOnOverlayClick = createElemPropsHook(usePopupModel)(model =
 
       if (
         dialog &&
+        modality !== 'touch' &&
```

Comment:
We use modality in this case because on touch devices, Modals should not be dismissed if you click on the over overlay. This is a behavior change which is why modality makes sense in this case.